geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java

```diff
@@ -0,0 +1,260 @@
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.apache.geode.management.internal.cli.result.ResultBuilder.buildResult;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import org.apache.logging.log4j.Logger;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.distributed.internal.ClusterConfigurationService;
+import org.apache.geode.internal.datasource.ConfigProperty;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.cli.CliMetaData;
+import org.apache.geode.management.cli.Result;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.functions.CreateJndiBindingFunction;
+import org.apache.geode.management.internal.cli.functions.JndiBindingConfiguration;
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.management.internal.cli.result.ResultBuilder;
+import org.apache.geode.management.internal.configuration.domain.Configuration;
+import org.apache.geode.management.internal.configuration.utils.XmlUtils;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class CreateJndiBindingCommand implements GfshCommand {
+  private static final Logger logger = LogService.getLogger();
+
+  static final String CREATE_JNDIBINDING = "create jndi-binding";
+  static final String CREATE_JNDIBINDING__HELP =
+      "Create a jndi binding that holds the configuration for the XA datasource.";
+  static final String BLOCKING_TIMEOUT_SECONDS = "blocking-timeout-seconds";
+  static final String BLOCKING_TIMEOUT_SECONDS__HELP =
+      "This element specifies the maximum time to block while waiting for a connection before throwing an exception.";
+  static final String CONNECTION_POOLED_DATASOURCE_CLASS = "conn-pooled-datasource-class";
```

Comment:
I'd prefer to not see "connection" spelled out completely in the option string rather than the abbreviation "conn". Abbreviations can lead to misinterpretations.

Reply:
Agreed, however unfortunately `conn-pooled-datasource-class` is a parameter defined by existing JNDI parameters so I think it would best to keep them consistent.

Reply:
`conn-pooled-datasource` is consistent with xml schema attributes.